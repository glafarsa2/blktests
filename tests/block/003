#!/bin/bash
#
# Run a range of discards on a device, anywhere from 4k to 4g
#
# Copyright (C) 2017 Jens Axboe
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.

TEST_GROUPS=(block auto)

prepare() {
	_have_fio
}

test_device() {
	echo "Running ${TEST_NAME}"

	fio --output=/dev/null --bsrange=4k-4g --rw=randtrim --norandommap \
		--name=discards --filename="$TEST_DEV" --runtime=10s --time_based=1

	echo "Test complete"
}
