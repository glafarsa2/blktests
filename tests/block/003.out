Running block/003
Test complete
